### ISSUE TITLE
Listing dbt models... never completes with high model coun

### ISSUE THREAD

**Issue by radian21** [innoverio/vscode-dbt-power-user#58]
I love this extension, but I found a flaw. My dbt project has 1,100+ models in it. When I reached some model count threshold, the part of the extension that lists (sees all) models started to hang. It seems that, as such, the extension doesn't register newly added models (above the threshold?) and the parent/child DAG relationship features don't work for these newer models.

On the VS Code status bar, it shows "Listing dbt models..." with an infinitely spinning wait cursor - the operation never completes. I've looked to see if there is a config setting for this, but there isn't.

![dbt extension error](https://user-images.githubusercontent.com/4933973/112527705-544bb280-8d71-11eb-92e0-6e7941d20778.png)

Is this a known bug? Solution? Work-around?

Thanks. Again, this is a really helpful extension!

Running:

Windows 10
VS Code 1.54.3
dbt 0.19.0
python 3.8.8 x64

**Comment by mdesmet** [comment]
under the hood, it would execute 'dbt ls'. It should wait until that one completes.

You can disable this behavior in the settings, see #38 , however it should still complete, there is no limit set in the extension. Could be of course some limit in vscode that I'm not aware of.

Can you include the logs, maybe i can deduce something else from there?

`cmd/ctrl-shift-p -> Search Show Logs -> Extension Host`

**Comment by radian21** [comment]
Thanks for the response.  I've included the vscode developer logs as requested.  I am able to `dbt ls` fine on the command line.  Does the extension assume a certain directory structure to traverse for the `ls` operation? My dbt project directories comply with the dbt basic structure, but I've introduced custom directories within that for code organization.

[vscode-dev-logs.txt](https://github.com/innoverio/vscode-dbt-power-user/files/6222282/vscode-dev-logs.txt)
